core/src/main/scala/contextual/contextual.scala

```diff
@@ -67,7 +67,7 @@ trait Interpolator extends Interpolator.Parts { interpolator =>
     val context: compat.Context = null
 
     /** The expressions that are substituted into the interpolated string. */
-    def expressions: Seq[context.Tree] = Nil
+    def expressions: Seq[context.Expr[Any]] = Nil
```

Comment:
Leave this as `context.Tree`, and call `.map(compat.toTree)` on the expressions value where necessary. You'll need to implement `compat.toTree` to call `.tree` on an `c.Expr` in 2.10, or just be the identity in 2.11/2.12.